src/java.base/share/classes/com/sun/crypto/provider/PBEKey.java

```diff
@@ -202,6 +205,9 @@ private void readObject(java.io.ObjectInputStream s)
     @java.io.Serial
     private Object writeReplace() throws java.io.ObjectStreamException {
         try {
+            if (isDestroyed()) {
+                throw new NotSerializableException("key is destroyed");
```

Comment:
nit: for uniformity, you may want to capitalize `key` since it is capitalized elsewhere

Reply:
Yes, will fix.